airflow/jobs/job.py

```diff
@@ -117,6 +115,15 @@ def __init__(self, executor=None, heartrate=None, **kwargs):
     def executor(self):
         return ExecutorLoader.get_default_executor()
 
+    @cached_property
+    def heartrate(self):
+        if self.job_type == "TriggererJob":
+            return conf.getfloat("triggerer", "JOB_HEARTBEAT_SEC")
+        else:
+            # Heartrate used to be hardcoded to scheduler, so in all other
+            # cases continue to use that value for back compat
+            return conf.getfloat("scheduler", "JOB_HEARTBEAT_SEC")
+
```

Comment:
I wonder if we can avoid implementing this logic in the parent class (job class), where doing so, we will have to add a case for each future job runner.

Currently for Triggerer Job, we pass the heartbeat as a parameter when we run the triggerer command:
https://github.com/apache/airflow/blob/56c41d460c3f2a4e871c7834033c3152e71f71d2/airflow/cli/commands/triggerer_command.py#L60-L61

Why do we need to override the default value too?

Reply:
> I wonder if we can avoid implementing this logic in the parent class (job class), where doing so, we will have to add a case for each future job runner.

Yeah and the interaction between jobs and their runners are very blurry (a job doesn't set it's own type even, it's the runner that does it!).

My aim with this PR was just to keep it scoped to the bug fix and not rearchitect too much.

> Why do we need to override the default value too?

I'm not sure precisely which default value you're talking about. But the heartrate needs to be set in the Job class because the heartrate is not stored in the DB for Jobs, so it needs to be fetched from cconfig when sqlalchemy builds the Python object from a query result.

Reply:
> But the heartrate needs to be set in the Job class because the heartrate is not stored in the DB for Jobs

Ah yes, I see the need to do this :thinking: